modules/database/src/ioc/dbStatic/dbLexRoutines.c

```diff
@@ -1167,8 +1167,28 @@ static void dbRecordField(char *name,char *value)
     pdbentry = ptempListNode->item;
     status = dbFindField(pdbentry,name);
     if (status) {
-        epicsPrintf("Record \"%s\" does not have a field \"%s\"\n",
-            dbGetRecordName(pdbentry), name);
+        epicsPrintf("%s Record \"%s\" does not have a field \"%s\"\n",
+                    dbGetRecordTypeName(pdbentry), dbGetRecordName(pdbentry), name);
+        if(dbGetRecordName(pdbentry)) {
```

Comment:
We should probably avoid passing a NULL `dbGetRecordName(pdbentry)` to `epicsPrintf()` (most libc implementations do handle that okay, but it doesn't look good). Move the print inside the conditional?

Reply:
I don't think it is actually possible to reach `dbRecordField()` unless the record exists.  I can't recall ever seeing `Record "(null)"`.  So the condition is I think mainly paranoia.  Also, I was anyway going to have a separate block for the added locals.